### PR TITLE
Bug 1159783 - [gaia-component] lint fix and add auto run lint command in travis

### PR DESCRIPTION
**Pull request by gasolin**
- use gaia .jshintrc
- fix lints and pass unit tests
- run lint in travis

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,4 +12,13 @@ before_script:
...
-script: "npm test"
+script:
+  - $CI_ACTION
+
+env:
+  global:
+    - TEST_SUITE=unit
+  matrix:
+    - CI_ACTION="npm run lint"
+    - CI_ACTION="npm test"
```
Comment (wilsonpage) [inline .travis.yml:23]: Can you explain how the above works? I'm not that familiar with Travis.

File: gaia-component.js [hunk 2]
```diff
@@ -160,9 +159,7 @@ var base = {
         if (this.lightStyle) { this.appendChild(this.lightStyle); }
       },
 
-      get: function() {
-        return textContent.get();
-      }
+      get: textContent.get
```
Comment (wilsonpage) [inline gaia-component.js]: I thought you said Chrome chokes on this?

Comment (gasolin) [inline gaia-component.js]: thanks for reminding that. I was wondering the issue is cause by extra `()`, but in real chrome test it still chokes, so I'd rollback the syntax.

File: .travis.yml (hunk 1 above)
Comment (gasolin) [inline .travis.yml:23]: The `matrix` is used to run multiple tasks in parallel http://docs.travis-ci.com/user/build-configuration/#The-Build-Matrix

we define a `$param` under script, and travis will take params from env > metrix to create tasks.
The `CI_ACTION` naming is referenced from old gaia travis.

We can even use matrix to run tests under different node.js version

```
node_js:
  - '0.12'
  - 'iojs'
  - '0.10'
```

Comment (wilsonpage) [inline .travis.yml:23]: Cool thanks!

**Pull request merged**